lib/puppet/provider/rds_instance/v2.rb

```diff
@@ -113,8 +113,15 @@ def create
       availability_zone: resource[:availability_zone],
     }
 
-    rds_client(resource[:region]).create_db_instance(config)
-
+    if resource[:restore_snapshot]
+      Puppet.info("Restoring DB instance #{name} from snapshot #{resource[:restore_snapshot]}")
+      [:engine_version, :backup_retention_period].each { |k| config.delete(k) }
```

Comment:
That is correct. The RDS engine version must be the same as the snapshot was created with.